app/scripts/directives/overview/mobileClientRow.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+(function () {
+  angular.module('openshiftConsole').component('mobileClientRow', {
+    controller: [
+      '$scope',
+      '$filter',
+      '$routeParams',
+      'ProjectsService',
+      'DataService',
+      'APIService',
+      'MobileClientsService',
+      'ListRowUtils',
+      'Navigate',
+      'AuthorizationService',
+      MobileAppRow,
+
+    ],
+    controllerAs: 'row',
+    bindings: {
+      apiObject: '<',
+      state: '<',
+      bindings: '<'
+    },
+    templateUrl: 'views/overview/_mobile-client-row.html'
+  });
+
+  function MobileAppRow($scope, $filter, $routeParams, ProjectsService, DataService, APIService, MobileClientsService, ListRowUtils, Navigate, AuthorizationService) {
+    var row = this;
+    $scope.installType = '';
+
+    _.extend(row, ListRowUtils.ui);
+    
+    row.bundleDisplay = "bundle id: " + row.apiObject.spec.appIdentifier;
+    switch (row.apiObject.spec.clientType) {
+      case 'android':
+        row.icon = 'fa fa-android';
+        $scope.installType = 'maven';
```

Comment:
Does `class="{{row.apiObject.metadata.labels.icon}}"` on the relevant element work?
If it doesn't creating a filter for this makes sense